src/psyclone/domain/nemo/transformations/nemo_arrayaccess2loop_trans.py

```diff
@@ -0,0 +1,328 @@
+        # Not a PSyIR node
+        if not isinstance(node, Node):
+            raise TransformationError(
+                "Error in NemoArrayAccess2LoopTrans transformation. The "
+                "supplied node argument should be a PSyIR Node, but found "
+                "'{0}'.".format(type(node).__name__))
+        # Not within an array reference
+        if not node.parent or not isinstance(node.parent, ArrayReference):
+            raise TransformationError(
+                "Error in NemoArrayAccess2LoopTrans transformation. The "
+                "supplied node argument should be within an ArrayReference "
+                "node, but found '{0}'.".format(type(node.parent).__name__))
+        array_ref = node.parent
+        # Array reference not within an assignment
+        if not array_ref.parent or not isinstance(array_ref.parent,
+                                                  Assignment):
+            raise TransformationError(
+                "Error in NemoArrayAccess2LoopTrans transformation. The "
+                "supplied node argument should be within an ArrayReference "
+                "node that is within an Assignment node, but found '{0}'."
+                .format(type(array_ref.parent).__name__))
+        assignment = array_ref.parent
+        # Array reference not on lhs of the assignment
+        if assignment.lhs is not array_ref:
+            raise TransformationError(
+                "Error in NemoArrayAccess2LoopTrans transformation. The "
+                "supplied node argument should be within an ArrayReference "
+                "node that is within the left-hand-side of an Assignment "
+                "node, but it is on the right-hand-side.")
+
+        # Contains a range node
+        if node.walk(Range):
+            raise TransformationError(
+                "Error in NemoArrayAccess2LoopTrans transformation. The "
+                "supplied node should not be or contain a Range node as it "
+                "should be single valued.")
+
+        # Capture loop iterator symbols in order
+        iterator_symbols = []
+        location = node.parent.parent
+        while (isinstance(location.parent, Schedule) and
+               isinstance(location.parent.parent, (Loop, NemoKern))):
+            location = location.parent.parent
+            if isinstance(location, Loop):
+                iterator_symbols.append(location.variable)
+
+        # Index contains a loop iterator
+        for reference in node.walk(Reference):
+            if reference.symbol in iterator_symbols:
+                raise TransformationError(
+                    "Error in NemoArrayAccess2LoopTrans transformation. The "
+                    "supplied node should not be or contain a loop iterator, "
+                    "it should be single valued.")
+
+        # An index before this one has more than one loop iterator
+        array_access_iterators = []
+        for index in node.parent.children[:node.position]:
```

Comment:
s/.children/.indices/ Oh, and `node.position` needs correcting too.

Reply:
This code has been removed in the restructuring.